### ISSUE TITLE
Rebooting the bare-metal this is deployed on results in bizarre routing

### ISSUE THREAD

**Issue by ShawnClake** [evertramos/docker-compose-letsencrypt-nginx-proxy-companion#103]
# Error description:
I'm using docker-compose in conjunction with this reverse proxy application to automate the deployment, routing, and encryption of a micro-service deployment. When I set this up all goes well. However, if I reboot the bare-metal server this is running on, the sub-domains will route to random containers. This error has haunted me for months and has made it so that I am unable to deploy this anywhere outside of a testing environment. 

To fix this, I use docker-compose stop on this repository as well as on my own stack. Then I remove the `data` directory which this repository utilizes. Then, if I start the reverse proxy, wait a few seconds, and boot my own stack everything works fine again.

# Steps to reproduce
* Deploy this reverse proxy
* Deploy your own stack using docker-compose
* Reboot the server these are running on

# Expected result
* db.mydomain.com -> Routes to MySQL
* pma.mydomain.com -> Routes to PHPMyAdmin
* mydomain.com -> Routes to the main web container

etc.

# Error result
* db.mydomain.com -> Routes to one of the containers in the stack. (ie. maybe this now routes to the main web container)
* pma.mydomain.com -> Routes to one of the containers in the stack. (ie. maybe this now routes to the MySQL DB)
* mydomain.com -> Routes to one of the containers in the stack. (ie. maybe this now routes to the PHPMyAdmin)

# Comments
* Here are the modifications I made to the reverse proxy container: https://github.com/No-Patience/Reverse-Proxy
* Here is the current deployment we use: https://github.com/No-Patience/No-Patience

**Comment by ShawnClake** [comment]
I also noticed that when I reboot the server, the nginx-gen container does not come back up.

**Comment by j7an** [comment]
I've noticed sometimes not all the primary containers (nginx, docker-gen, and letsencrypt) start up as well. @ShawnClake, can you try using Docker's [`depends_on`](https://docs.docker.com/compose/startup-order/) statement to see if it makes a difference for you?

A more advanced option is to [wait for another container to be ready before starting another container](https://medium.com/@edgar/how-to-wait-for-a-container-to-be-ready-before-starting-another-container-using-docker-compose-92bab2fc1633).

**Comment by ShawnClake** [comment]
Hi @j7an, which order should they be starting in? Is the order of the startup the cause of the issue I'm experiencing?

**Comment by ShawnClake** [comment]
Thanks, I should have a few hours this weekend I can spend tinkering with this. I'll reply back once I know more.

**Comment by ShawnClake** [comment]
Update: I re-deployed the app fresh and with that came an update to the nginx-web and nginx-gen containers. I've yet to reproduce this issue on that fresh deployment. I'm not sure whether that means this is resolved now or that I just haven't gotten unlucky yet.

**Comment by StefanFransen** [comment]
Hey I've got 2 websited deployed with their own containers.
When I visit one of the websites it rotates between the two containers.

Request 1: Container 1 loads
Request 2: Container 2 loads
Request 3: Container 1 loads

This is after deploying the containers, the server hasn't rebooted yet.

**Comment by evertramos** [comment]
Hello @ShawnClake,

Have you tried that recently? I have tested this in the past two weeks, one of the test with production environment with 36 containers running, 14 websites, and databases, and it did came up everything accordingly, so, please update.

Closing this issue... if you face this same problem, please reopen it!

Thanks!

**Comment by chrisckc** [comment]
I have experienced this when setting all 3 containers to "restart: unless-stopped"
after restarting docker , only the  "jrcs/letsencrypt-nginx-proxy-companion" container would be running, the other 2 failed to start, presumably due to incorrect startup order.

I tried "depends_on", but it is just a docker-compose thing, it is not used by docker when it starts containers by itself after a reboot.

The interesting thing is that if i use the default setting in the repo's compose config of "restart: always" it seems to work every time, i have restarted docker several times and all 3 containers startup fine.

According to the Docker docs, in this scenario there should be difference between "always" and "unless-stopped" , restarting docker should still auto start the containers, unless they have been stopped using "docker stop".
